### ISSUE TITLE
DetectoRS with ResNeXt-101-32x4d backbone for Instance Segmentation

### ISSUE THREAD

**Issue by borgarpa** [open-mmlab/mmdetection#3951]
Hi,

I am trying to use DetectoRS model for Instance Segmentation task. However, I want to implement the ResNeXt-101 32x4d backbone version, which is the current state-of-the-art in COCO test-dev dataset. 

The thing is I don't exactly know how to implement it. I have tried using the following config modification:

```
_base_ = '../htc/htc_x101_32x4d_fpn_16x1_20e_coco.py'

model = dict(
    backbone=dict(
        type='DetectoRS_ResNeXt',
        conv_cfg=dict(type='ConvAWS'),
        sac=dict(type='SAC', use_deform=True),
        stage_with_sac=(False, True, True, True),
        output_img=True),
    neck=dict(
        type='RFP',
        rfp_steps=2,
        aspp_out_channels=64,
        aspp_dilations=(1, 3, 6, 1),
        rfp_backbone=dict(
            rfp_inplanes=256,
            type='DetectoRS_ResNeXt',
            depth=101,
            num_stages=4,
            out_indices=(0, 1, 2, 3),
            frozen_stages=1,
            norm_cfg=dict(type='BN', requires_grad=True),
            norm_eval=True,
            conv_cfg=dict(type='ConvAWS'),
            sac=dict(type='SAC', use_deform=True),
            stage_with_sac=(False, True, True, True),
            pretrained='open-mmlab://resnext101_32x4d',
            style='pytorch')))
```

However, I'm struggling to make it work for Instance Segmentation. I am bit of a newbie using MMDetection, so I would greatly appreciate any help you could provide.

**Environment**

```
sys.platform: linux
Python: 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
CUDA available: True
GPU 0: Tesla T4
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.6.0+cu101
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.5.0 (Git Hash e2ac1fac44c5078ca927cb9b90e1b3066a0b2ed0)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.7.0+cu101
OpenCV: 4.1.2
MMCV: 1.1.5
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.5.0+5d07dda
```

**Comment by v-qjqs** [comment]
Hi @borgarpa, you might specify the issues or messages you got.

**Comment by borgarpa** [comment]
Hi,

I tried to download the weights from Siyuan Qiao page: [joe-siyuan-qiao/DetectoRS](https://github.com/joe-siyuan-qiao/DetectoRS). The url I used was the following: [https://www.cs.jhu.edu/~syqiao/DetectoRS/DetectoRS_X101-ed983634.pth](https://www.cs.jhu.edu/~syqiao/DetectoRS/DetectoRS_X101-ed983634.pth).

The full configuration of the model is as follows:
[detectors_configs.txt](https://github.com/open-mmlab/mmdetection/files/5404247/detectors_configs.txt)

When initializing the model with `mmdet.apis.init_detector` function and the configs and checkpoint specified above, the following shows up:
[detectors_101.txt](https://github.com/open-mmlab/mmdetection/files/5404234/detectors_101.txt)

Then, when performing inference, no mask is drawn. That is, no object is detected. Hence, my questions are:
- Is my config file right? Should the model work as expected?
- If the configs are wrong, what should I change?
- Since the checkpoint is downloaded from [joe-siyuan-qiao/DetectoRS](https://github.com/joe-siyuan-qiao/DetectoRS) repository, and it was implemented on a previous version of mmdet, should it work? Or should I re-train the model in order to generate a valid checkpoint file for the ResNeXt-101 32x4d version?

Thank you for your time and patience!

**Comment by jyuan1118** [comment]
Hi, 
  
  Is your pretrained model does not match your configs file? 

   I am also facing some re-implement problem. 

   I was tried to re-implement the detectors_htc_r50_1x_coco.py recently. But the segm_mAP is very low. It stops around 0.275 after 5 epochs. I did not modified anything and just set the batch size to 1 due to the capacity of GPU memories. Did you know what's the wrong with it?

  Thank you for your time .

**Comment by vpeopleonatank** [comment]
> Hi,
> 
> Is your pretrained model does not match your configs file?
> 
> I am also facing some re-implement problem.
> 
> I was tried to re-implement the detectors_htc_r50_1x_coco.py recently. But the segm_mAP is very low. It stops around 0.275 after 5 epochs. I did not modified anything and just set the batch size to 1 due to the capacity of GPU memories. Did you know what's the wrong with it?
> 
> Thank you for your time .

I 've faced the same issue, but after put pretrained model in `load_from`, model seems converge faster.

**Comment by dimimal** [comment]
Is there any update wrt training with ReseNext101?

**Comment by borgarpa** [comment]
> 
> 
> Is there any update wrt training with ReseNext101?

I was wondering the same... I am still trying to make it work.

**Comment by AronLin** [comment]
For x101_32x4d, you should set `groups=32` and `base_width=4` in `RFP`, here is the config file:

    _base_ = '../htc/htc_x101_32x4d_fpn_16x1_20e_coco.py'

    model = dict(
        backbone=dict(
            type='DetectoRS_ResNeXt',
            conv_cfg=dict(type='ConvAWS'),
            sac=dict(type='SAC', use_deform=True),
            stage_with_sac=(False, True, True, True),
            output_img=True),
        neck=dict(
            type='RFP',
            rfp_steps=2,
            aspp_out_channels=64,
            aspp_dilations=(1, 3, 6, 1),
            rfp_backbone=dict(
                rfp_inplanes=256,
                type='DetectoRS_ResNeXt',
                depth=101,
                groups=32,
                base_width=4,
                num_stages=4,
                out_indices=(0, 1, 2, 3),
                frozen_stages=1,
                norm_cfg=dict(type='BN', requires_grad=True),
                norm_eval=True,
                conv_cfg=dict(type='ConvAWS'),
                sac=dict(type='SAC', use_deform=True),
                stage_with_sac=(False, True, True, True),
                init_cfg=dict(
                    type='Pretrained', checkpoint='open-mmlab://resnext101_32x4d'),
                style='pytorch')))
    
    optimizer_config = dict(
        _delete_=True, grad_clip=dict(max_norm=35, norm_type=2))

**Comment by AronLin** [comment]
The result:
bbox mAP: 51.5
mask mAP: 44.7
